Source/TestDafny/README.md

```diff
@@ -0,0 +1,37 @@
+# TestDafny (a.k.a %testdafny)
+
+This internal, testing-only utility is used to execute a single Dafny source file against each compile target option and assert that the output
+is the same as the corresponding `<test file>.expect` file. This is a very common testing approach implemented manually across the integration test suite with multiple
+lit commands. See [this test file](../../Test/metatests/ConsistentWhenSupported.dfy) for a comparison of the manual approach
+compared to using this utility.
+
+Note the manual approach has several downsides:
+
+1. New compilers have to be manually added to every relevant test file.
+2. The expected output must be duplicated for each compile target, making it hard to perceive inconsistencies.
+3. When a test fails, it is difficult to determine from the output which compile target failed.
```

Comment:
Not with the above, I think?

Same if we added `Dafny /compileTarget:all`; we would print `Running xyz.java`, `running xyz.cs`, etc.

Reply:
Yup, there are definitely other solutions to #2 and #3. I added a bit after the list to acknowledge that.